api/v1beta2/awscluster_webhook.go

```diff
@@ -174,6 +133,55 @@ func (r *AWSCluster) ValidateUpdate(old runtime.Object) (admission.Warnings, err
 	return nil, aggregateObjErrors(r.GroupVersionKind().GroupKind(), r.Name, allErrs)
 }
 
+func (r *AWSCluster) validateControlPlaneLoadBalancerUpdate(oldlb, newlb *AWSLoadBalancerSpec) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// TODO(nrb): The validate-on-update default test was failing when newlb was nil
+	// Upon investigation, both values were nil, which is unlikely in a real world scenario
+	if oldlb == nil && newlb == nil {
+		return nil
+	}
```

Comment:
```
newLoadBalancer := &AWSLoadBalancerSpec{}
	existingLoadBalancer := &AWSLoadBalancerSpec{}

	if r.Spec.ControlPlaneLoadBalancer != nil {
		newLoadBalancer = r.Spec.ControlPlaneLoadBalancer.DeepCopy()
	}

	if oldC.Spec.ControlPlaneLoadBalancer != nil {
		existingLoadBalancer = oldC.Spec.ControlPlaneLoadBalancer.DeepCopy()
	}
```

This was previously handled this way

Reply:
Which to be honest, maybe better to restructure, given that it was a bit confusing